osu.Game/Online/Rooms/PlaylistExtensions.cs

```diff
@@ -10,6 +13,21 @@ namespace osu.Game.Online.Rooms
 {
     public static class PlaylistExtensions
     {
+        /// <summary>
+        /// Returns the first non-expired <see cref="PlaylistItem"/> in playlist order from the supplied <paramref name="playlist"/>,
+        /// or the last-played <see cref="PlaylistItem"/> if all items are expired,
+        /// or <see langword="null"/> if <paramref name="playlist"/> was empty.
+        /// </summary>
+        public static PlaylistItem? GetCurrentItem(this ICollection<PlaylistItem> playlist)
+        {
+            if (playlist.Count == 0)
+                return null;
+
+            return playlist.All(item => item.Expired)
```

Comment:
It's not a huge deal but I would also add a default case. For example, let's suppose that Playlists is updated so that upon a room expiring, all items from the room are expired too. I'd probably return `.FirstOrDefault()` to cover that scenario.

As an aside, how about writing this in a similar way to osu-server-spectator:
https://github.com/ppy/osu-server-spectator/blob/0512ffee735c7a91693ce0776ad280669fd366bb/osu.Server.Spectator/Hubs/MultiplayerQueue.cs#L240-L241
Such that there's a `GetUpcomingItems()` and `GetHistoricalItems()` (or something like that) which could also be used for `MultiplayerQueueList` and `MultiplayerHistoryList`.

Reply:
I'm a little confused. Is this edge case not handled already? As in, with the code as it is, when all items are expired, the last played expired item will be returned. I'm not understanding how is that different from the scenario you mention.

Reply:
`PlaylistOrder` is not valid for expired items. The `PlaylistItem` member is missing the doc but it's present on `MultiplayerPlaylistItem`:
https://github.com/ppy/osu/blob/2ea31c26a58ddc1d954a02e5313057352b968186/osu.Game/Online/Rooms/MultiplayerPlaylistItem.cs#L42-L50

This is because at one point the server used to change the order for expired items too, though that doesn't seem to be the case anymore. Still prefer using `PlayedAt`.

Reply:
I've made some changes and introduced the two new extension methods as suggested. Let me know if they meet your expectations.